projects/prod_test/ClockI2CCommands.c

```diff
@@ -0,0 +1,406 @@
+/*
+ * ClockI2CCommands.c
+ *
+ *  Created on: January 10, 2025
+ *      Author: mcoshiro
+ *
+ * Contains code for MCU to clock synth I2C tests used for Apollo CM production
+ * tests
+ */
+
+// includes for types
+#include <stdbool.h>
+#include <stdint.h>
+
+// FreeRTOS
+#include "FreeRTOS.h"
+#include "task.h"
+
+// local includes
+#include "common/smbus.h"
+#include "common/pinsel.h"
+#include "common/printf.h"
+#include "common/utils.h"
+#include "commands.h"
+#include "InterruptHandlers.h"
+#include "ClockI2CCommands.h"
+#include "I2CCommunication.h"
+
+// device info
+struct dev_moni2c_addr_t clk_moni2c_addrs[NDEVICES_CLK] = {
+    {"r0a", U84_ADDR, R0A_MUX_BIT, SI5395_I2C_ADDR},
+    {"r0b", U84_ADDR, R0B_MUX_BIT, SI5395_I2C_ADDR},
+    {"r1a", U84_ADDR, R1A_MUX_BIT, SI5395_I2C_ADDR},
+    {"r1b", U84_ADDR, R1B_MUX_BIT, SI5395_I2C_ADDR},
+    {"r1c", U84_ADDR, R1C_MUX_BIT, SI5395_I2C_ADDR},
+};
+struct dev_ioexpander_addr_t ioexpander_addrs[NDEVICES_CLK_IOEXPANDER] = {
+    {U84_ADDR, U88_MUX_BIT, U88_ADDR},
+    {U84_ADDR, U83_MUX_BIT, U83_ADDR},
+};
+
+/**
+ * @brief Runs 1st stage of clock I2C communication test, testing synth ICs
+ *
+ * @details
+ * Performs a loop over clock synth ICs where some (distinct) data is written
+ * to a user scratch register on each IC, then a second loop reads the data and
+ * verifies that it matches what was written
+ *
+ * @param [out] m  output string
+ * @return true if test passes, false otherwise
+ */
+bool run_clock_i2ctest_synth(char *m)
```

Comment:
Maybe make clear which of these are CLI functions and which are helper functions.